theano/gpuarray/tests/test_opt.py

```diff
@@ -467,30 +468,20 @@ def test_many_arg_elemwise():
 
                 outputs = []
                 for mode in [mode_with_gpu, mode_without_gpu]:
-                    # test the optijmization local_gpu_elemwise_0
+                    # test the optimization local_gpua_elemwise
                     f = theano.function(
-                        symb_args, op_to_test(*symb_args),
-                        mode=mode.excluding("local_gpu_elemwise_1"))
+                        symb_args, op_to_test(*symb_args))
                     outputs.append(f(*args))
-                    # assert that the test was done on the gpu.
-                    if mode is mode_with_gpu:
-                        assert any([isinstance(node.op, GpuElemwise)
-                                    for node in f.maker.fgraph.apply_nodes])
 
-                    # test the optijmization local_gpu_elemwise_1
-                    f = theano.function(
-                        symb_args,
-                        GpuFromHost(test_ctx_name)(op_to_test(*symb_args)),
-                        mode=mode.excluding("local_gpu_elemwise_0"))
-                    out = f(*args)
```

Comment:
Why all those changes to the test? It lower the test quality. You don't call the compiled function anymore. You don't test each optimization individually as before.